src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_INCRICE = 1.5;
+    private int size;
+    private int capacity;
+    private Object[] elementT;
+
+    public ArrayList() {
+        this.elementT = new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
+
+    public void resize() {
+        capacity = (int) (capacity * CAPACITY_INCRICE);
+        T[] tempArray = (T[]) new Object[capacity];
+        System.arraycopy(elementT, 0, tempArray, 0, size);
+        elementT = tempArray;
+    }
+
+    public int getIndexOf(T element) {
+        for (int i = 0; i < size; i++) {
+            if ((elementT[i] == null && element == null)
+                    || (Objects.equals((T) elementT[i], element))) {
+                return i;
+            }
+        }
+        throw new NoSuchElementException("Can't found element by value");
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element on position " + index);
+        }
+        if (size == capacity) {
+            resize();
+        }
+        T[] tempArray = (T[]) new Object[capacity];
+        System.arraycopy(elementT, 0, tempArray, 0,index);
```

Comment:
here you can do with one `.arraycopy()` (copy from `elemenT` to `elemenT` all elements after the `index`, shifting them by 1 to the left (this can be achieved by specifying `index` in srcPos and `index+1` in destPos) and after `.arraycopy()` put the value you need in `element[index]`)